GitUI/UserControls/RevisionGrid/Columns/CommitIdColumnProvider.cs

```diff
@@ -66,8 +67,26 @@ public override bool TryGetToolTip(DataGridViewCellMouseEventArgs e, GitRevision
                 return false;
             }
 
-            toolTip = revision.Guid;
+            toolTip = Format(revision.Guid);
             return true;
         }
+
+        public static string Format(string guid)
+        {
+            var text = new StringBuilder();
+            for (int pos = 0, length = guid.Length; pos < length; pos += 4)
```

Comment:
why not 
for (int pos = 0; pos < guid.Length; pos += 4)

Reply:
I do not know and shall not know the implementation of the *property* string.Length. It might add some overhead getting it multiple times.
I am afraid the compiler/optimizer might not recognize that `guid` and its length is constant. (C# has only a limited concept of constant/read-only expressions.)
Code for a single tooltip is not time-critical. Though I prefer to always consider the performance.
But if you tell me that it is not necessary (at least for string), I will use the shorter implementation and will omit such helper variables.

Reply:
comma operator is a little obscure (I thought it was not even allowed in C# for())
I do not know how the compiler optimizes this, but would not consider this critical anyway and prefer readability. Not a strong opinion.

Reply:
I think you're trying to optimise prematurely here. .NET strings are immutable, so a string's length is a constant.

Slight OT, most of the heavy optimisation lifting these days is done by the JIT, and it is pretty smart.
This article may be useful: http://benhall.io/c-debug-vs-release-builds-and-debugging-in-visual-studio-from-novice-to-expert-in-one-blog-article/